SSA/Projects/LLVMAndRiscv.lean

```diff
@@ -125,4 +127,184 @@ attribute [simp, simp_denote] outTy_map_signature_eq
+              have hcontra: List.length Γ ≤ v.val → Γ.get? v.val  = none := by simp [List.get?]
+              have ll: (ctxtTransformToRiscV Γ).length = Γ.length := by
+                  unfold ctxtTransformToRiscV Ctxt.map
+                  conv =>
+                  rw [List.length_map]
+              rw [← ll]
+              by_contra x
+              simp only [RISCV64.RV64, Ctxt.get?.eq_1, gt_iff_lt, not_lt] at x
+              rw [← ll] at hcontra
+              apply hcontra at x
+              have hh : Γ.get? v.val = none → (ctxtTransformToRiscV Γ).get? v.val = none := by
+                simp only [LLVMPlusRiscV, Ctxt.get?, RISCV64.RV64, Ctxt.get?.eq_1,
+                  getElem?_eq_none_iff, not_lt]
+                intros x
+                rw [ll]
+                exact x
+              apply hh at x
+              rw [x] at hv
+              contradiction
+            ⟩ with
+            | .llvm _ =>
+                throw <| .generic s!"INTERNAL ERROR: This case is impossible, RISCV expression is pointing to LLVM variable."
+            | .riscv originalRISCVTy =>
+                if hty' : originalRISCVTy = t then
+                  return ⟨v.val, by
+                  rw [← hty']
+                  simp only [LLVMPlusRiscV, Ctxt.get?,
+                    RISCV64.RV64, List.getElem?_eq_some_iff]
+                  simp only [LLVMPlusRiscV, RISCV64.RV64,
+                     Ctxt.get?, List.get_eq_getElem] at h
+                  rw [← Option.some.injEq] at h
+                  simp only [Option.some.injEq] at h
+                  have h1 := v.2
+                  simp only [RISCV64.RV64, Ctxt.get?] at h1
+                  rw [← List.get?_eq_getElem? ] at h1
+                  have ⟨bound, val⟩ := List.get?_eq_some_iff.mp h1
+                  have ll: (ctxtTransformToRiscV Γ).length = Γ.length := by
+                    unfold ctxtTransformToRiscV Ctxt.map
+                    conv =>
+                      rw [List.length_map]
+                  rw [ll] at bound
+                  use bound
+                  ⟩
+                else
+                  throw <|.generic s!"INTERNAL ERROR: This case is impossible, RISCV expression is pointing to an incorrect bitwidth LLVM argument."
+        return Expr.mk
+          (op := Op.riscv op1)
+          (eff_le := eff_le1)
+          (ty_eq := ty_eq1 ▸ rfl)
+          (args := args')
+          (regArgs := HVector.nil)
+
+instance : MLIR.AST.TransformTy LLVMPlusRiscV 0 where
+  mkTy tStx := do
+  try
+    let llvmParse ← InstcombineTransformDialect.mkTy tStx
+    return .llvm llvmParse
+  catch _llvmErr =>
+    try
+      let riscvParse ← RiscvMkExpr.mkTy tStx
+      return .riscv riscvParse
+    catch _riscvErr =>
+        throw <|.generic s!" INTERNAL ERROR : While trying to transform from MLIR AST to dialect specific AST
+         the transformation failed. The errors thrown are:
+          s!{(toString (repr _riscvErr))} and s!{(toString (repr _llvmErr ))}"
```

Comment:
```suggestion
  catch llvmErr =>
    try
      let riscvParse ← RiscvMkExpr.mkTy tStx
      return .riscv riscvParse
    catch riscvErr =>
        throw <|.generic s!" INTERNAL ERROR : While trying to transform from MLIR AST to dialect specific AST
         the transformation failed. The errors thrown are:
          s!{(toString (repr riscvErr))} and s!{(toString (repr llvmErr ))}"
```

If you use the variables, they shouldn't get a `_` before them!

Also, is this truly an INTERNAL ERROR? See the last comment, if this error can happen just because of bad input, it isn't an internal error, just a regular error.